tutordiscovery/templates/discovery/apps/settings/partials/common.py

```diff
@@ -20,12 +20,9 @@
     }
 }
 
-HAYSTACK_CONNECTIONS["default"].update(
-    {
-        "URL": "http://{{ ELASTICSEARCH_HOST }}:{{ ELASTICSEARCH_PORT }}",
-        "INDEX_NAME": "{{ DISCOVERY_INDEX_NAME }}",
-    }
-)
+ELASTICSEARCH_DSL['default'].update({
+    'hosts': "http://{{ ELASTICSEARCH_HOST }}:{{ ELASTICSEARCH_PORT }}/"
```

Comment:
What happens to the elasticsearch `INDEX_NAME`? I see that the index configuration is much more complex than it used to be. Should we simply deprecate the `DISCOVERY_INDEX_NAME` tutor setting?

Reply:
In lilac, index for course, course_run, person, program are separated. Simply deprecate DISCOVERY_INDEX_NAME works, but it may be useful to make indices configurable.

Reply:
> but it may be useful to make indices configurable.

I think this might be overkill: customising indices is probably an edge case. Do you personally need this feature? If not, let's remove the `DISCOVERY_INDEX_NAME` variable. The tutor discovery settings already include `{{ patch(...) }}` statement, so anyone who wants to override the setting is able to with a rather simple plugin.

Reply:
Sure, it can be done through plugin. BTW, the scenario I need to customize index name is when I want to separate indices for different instances while still share one ES cluster.

Reply:
If your company is a use case where indices need to be customised, then we can assume that this is a more widespread scenario. Can we create a single variable for all index overrides? I'm thinking something similar to:

```
INDEX_OVERRIDES =  {}
```

and then:

```
{% for name, index in DISCOVERY_INDEX_OVERRIDES.items() %}
ELASTICSEARCH_INDEX_NAMES["{{ name }}"] = "{{ index }}"
{% endfor %}
```

What do you think?

Reply:
Agree, that will be clearer.

Reply:
Can you please add a note to the README to explain this feature?